src/generate/CodeModelAzImpl.ts

```diff
@@ -2973,6 +2976,15 @@ export class CodeModelCliImpl implements CodeModelAz {
                 // else {
                 //     paramName = "name";
                 // }
+                else if (this.extensionName == 'vm') {
+                    if (paramName == 'vm_name') paramName = 'name';
+                } else if (this.extensionName == 'network') {
+                    if (paramName == 'network_security_group_name') paramName = 'nsg_name';
+                    else if (paramName == 'virtual_network_name') paramName = 'vnet_name';
+                    else if (commandGroup.toLowerCase() === 'privatelinkservices') {
+                        if (paramName == 'service_name') paramName = 'name';
+                    }
+                }
```

Comment:
Wondering why do we have this part? is this because of cli existing command name and swagger name mapping? or something else? What if we found more mismatch? Should we put it into a configurable file ?

Reply:
this is a quick fix for findings in depth coverage. they are existing param hard rule in CLI. I think they are similar with the mapping resource_group_name --> resource_group.

 I think cli-directive is a solution to configure this. But the problem will become who should configure these hard rules.

Reply:
I don't see the logic of handling this mapping part now?

Reply:
yes, handle this in RP's readme.az.md with cli-directive